src/pages/RFOX/components/Unstake/UnstakeConfirm.tsx

```diff
@@ -53,14 +117,169 @@ export const UnstakeConfirm: React.FC<UnstakeRouteProps> = () => {
         flex={1}
         mx={-2}
       >
-        <AssetIcon size='sm' assetId={asset?.assetId} />
+        <AssetIcon size='sm' assetId={stakingAsset?.assetId} />
         <Stack textAlign='center' spacing={0}>
-          <Amount.Crypto value='0.0' symbol={asset?.symbol} />
+          <Amount.Crypto value={unstakingAmountCryptoPrecision} symbol={stakingAsset?.symbol} />
           <Amount.Fiat fontSize='sm' color='text.subtle' value='0.0' />
         </Stack>
       </Card>
     )
-  }, [asset])
+  }, [stakingAsset, unstakingAmountCryptoPrecision])
+
+  const callData = useMemo(() => {
+    if (!stakingAsset) return
+
+    return encodeFunctionData({
+      abi: foxStakingV1Abi,
+      functionName: 'unstake',
+      args: [BigInt(toBaseUnit(unstakingAmountCryptoPrecision, stakingAsset.precision))],
+    })
+  }, [stakingAsset, unstakingAmountCryptoPrecision])
+
+  const isGetUnstakeFeesEnabled = useMemo(
+    () =>
+      Boolean(
+        stakingAssetAccountNumber !== undefined &&
+          wallet &&
+          stakingAsset &&
+          callData &&
+          feeAsset &&
+          feeAssetMarketData,
+      ),
+    [stakingAssetAccountNumber, wallet, stakingAsset, callData, feeAsset, feeAssetMarketData],
+  )
+
+  const {
+    data: unstakeFees,
+    isLoading: isUnstakeFeesLoading,
+    isSuccess: isUnstakeFeesSuccess,
+  } = useQuery({
+    ...reactQueries.common.evmFees({
+      to: RFOX_PROXY_CONTRACT_ADDRESS,
+      from: stakingAssetAccountAddress,
+      accountNumber: stakingAssetAccountNumber!, // see isGetStakeFeesEnabled
```

Comment:
Same comment here as the previous PR if you find a nice way to add type safety.

Reply:
TBD in https://github.com/shapeshift/web/issues/6973 after bumping react-query to latest